web/pdf_viewer.js

```diff
@@ -135,833 +133,825 @@ var PDFViewer = (function pdfViewer() {
+  }
 
+  /**
+   * @param {number} val - The page number.
+   */
+  set currentPageNumber(val) {
+    if ((val | 0) !== val) { // Ensure that `val` is an integer.
+      throw new Error('Invalid page number.');
+    }
+    if (!this.pdfDocument) {
+      return;
+    }
+    // The intent can be to just reset a scroll position and/or scale.
+    this._setCurrentPageNumber(val, /* resetCurrentPageView = */ true);
+  }
+
+  /**
+   * @private
+   */
+  _setCurrentPageNumber(val, resetCurrentPageView = false) {
+    if (this._currentPageNumber === val) {
       if (resetCurrentPageView) {
         this._resetCurrentPageView();
       }
-    },
-
-    /**
-     * @returns {string|null} Returns the current page label,
-     *                        or `null` if no page labels exist.
-     */
-    get currentPageLabel() {
-      return this._pageLabels && this._pageLabels[this._currentPageNumber - 1];
-    },
-
-    /**
-     * @param {string} val - The page label.
-     */
-    set currentPageLabel(val) {
-      let pageNumber = val | 0; // Fallback page number.
-      if (this._pageLabels) {
-        let i = this._pageLabels.indexOf(val);
-        if (i >= 0) {
-          pageNumber = i + 1;
-        }
-      }
-      this.currentPageNumber = pageNumber;
-    },
-
-    /**
-     * @returns {number}
-     */
-    get currentScale() {
-      return this._currentScale !== UNKNOWN_SCALE ? this._currentScale :
-                                                    DEFAULT_SCALE;
-    },
-
-    /**
-     * @param {number} val - Scale of the pages in percents.
-     */
-    set currentScale(val) {
-      if (isNaN(val)) {
-        throw new Error('Invalid numeric scale');
-      }
-      if (!this.pdfDocument) {
-        return;
-      }
-      this._setScale(val, false);
-    },
-
-    /**
-     * @returns {string}
-     */
-    get currentScaleValue() {
-      return this._currentScaleValue;
-    },
-
-    /**
-     * @param val - The scale of the pages (in percent or predefined value).
-     */
-    set currentScaleValue(val) {
-      if (!this.pdfDocument) {
-        return;
-      }
-      this._setScale(val, false);
-    },
-
-    /**
-     * @returns {number}
-     */
-    get pagesRotation() {
-      return this._pagesRotation;
-    },
-
-    /**
-     * @param {number} rotation - The rotation of the pages (0, 90, 180, 270).
-     */
-    set pagesRotation(rotation) {
-      if (!(typeof rotation === 'number' && rotation % 90 === 0)) {
-        throw new Error('Invalid pages rotation angle.');
-      }
-      if (!this.pdfDocument) {
-        return;
-      }
-      this._pagesRotation = rotation;
+      return;
+    }
 
-      for (let i = 0, ii = this._pages.length; i < ii; i++) {
-        let pageView = this._pages[i];
-        pageView.update(pageView.scale, rotation);
-      }
+    if (!(0 < val && val <= this.pagesCount)) {
+      console.error(
+        'PDFViewer._setCurrentPageNumber: "${val}" is out of bounds.');
```

Comment:
D'oh, this is embarrassing!
Thank you for spotting this; and as usual thanks for the review!